### PR TITLE
[DEM] Add bonds to discontinuum

### PR DESCRIPTION
**Pull request by joaquinirazabal**
In discontinuum calculations the force shown in the bond is the total force, not the elastic force as in the continuum case. Stresses are not printed.

### CODE REVIEW

File: applications/DEM_application/custom_elements/spheric_particle.h [hunk 1]
```diff
@@ -280,14 +282,15 @@ std::vector<array_1d<double, 3> > mNeighbourElasticExtraContactForces;
...
+virtual void CalculateOnContactElements(size_t i_neighbour_count, double LocalContactForce[3]);
 
 array_1d<double, 3> mContactMoment; //SLS
 
 Matrix* mStressTensor;
 Matrix* mSymmStressTensor;
 double mPartialRepresentativeVolume;
 
-
+unsigned int mNeighborsSize;
```
Comment (maceligueta) [inline applications/DEM_application/custom_elements/spheric_particle.h]: Can we avoid this variable by using `mNeighbors.size()` ?

Comment (joaquinirazabal) [reply applications/DEM_application/custom_elements/spheric_particle.h]: Of course we can, good point @maceligueta

**Comment by maceligueta** [comment]
Thanks, @joaquinirazabal ! You have worked with the code lately and probably you are the one to ask: we could create the bonds and put values on them in those time steps when we have to print? It's a problem because in C++ we do not know this information, but we could pass it from Python via ProcessInfo, right?
We have the same problem with the steps in which we need to search for neighbours...

**Comment by farrufat-cimne** [comment]
Wouldnt it be a good idea to use a pattern similar to the one used in the landedParticlesOperations call?: for example, 

from python
if is_time_to_print_bonds:
                self.landed_particles_utilities.MarkAndSaveLandedParticles()

and do whatever is required from a function in the cpp? in this case the function that create and write the values in the bonds.

**Comment by maceligueta** [comment]
In this case we cannot do it exactly like that. Because we need to write on the bonds during the loop over the neighbours, which is done in C++. So, some parts could be done as you say, but not all of them...

**Comment by joaquinirazabal** [comment]
Ok, I will try to implement that solution.

**Comment by joaquinirazabal** [comment]
I did it adding a new variable to process info called IS_TIME_TO_PRINT

**Comment by joaquinirazabal** [comment]
Do you know how to avoid the Codacy warning? If used_bonds_counter is declared inside the loop the variable is private and a segmentation fault appears...

**Comment by maceligueta** [comment]
Let's ignore Codacy for this time. I think it is a false positive. We want this variable to be shared by all threads, so it's ok if it's declared outside the parallel region.

**Comment by maceligueta** [comment]
I observed that you modified the API of `UpdateTimeInModelParts`, so I tagged the PR as API breaker. This is dangerous usually, but I can't think of a better way of achieving the new behaviour (well, we could move that function to the main_script.py, but then we would change the behaviour in a different sense. All in all, I think everything is fine. Approving

**Review by maceligueta — APPROVED** [review]

**Comment by joaquinirazabal** [comment]
Thanks, @maceligueta. Merged.

**Pull request merged**